### PR TITLE
Handle etcd custom uid/gid

### PR DESCRIPTION
**Pull request by moelsayed**

### CODE REVIEW

File: cert-deployer [hunk 1]
```diff
@@ -23,3 +23,10 @@ for i in $(env | grep -o KUBECFG_.*=); do
     chmod 600 $SSL_CRTS_DIR/$name
   fi
 done
+
+
+if [ ! -z ${ETCD_UID} ] && [ ! -z ${ETCD_GID} ]; then
+  setfacl -R -m m::rX ${SSL_CRTS_DIR}
```
Comment (superseb) [inline cert-deployer]: Adding this adds a dependency on ACL which is usually available but we see the weirdest setup, can we check if it succeeds and else log an error? Can we also add a comment describing what we are doing here so people without Linux ACL knowledge know whats happening (or knows what the intention is)

File: cert-deployer [hunk 2]
```diff
@@ -23,3 +23,10 @@ for i in $(env | grep -o KUBECFG_.*=); do
     chmod 600 $SSL_CRTS_DIR/$name
   fi
 done
+
+
+if [ ! -z ${ETCD_UID} ] && [ ! -z ${ETCD_GID} ]; then
```
Comment (superseb) [inline cert-deployer]: `-n` is the opposite of `-z` and can we quote the strings

File: cert-deployer [hunk 3]
```diff
@@ -23,3 +23,10 @@ for i in $(env | grep -o KUBECFG_.*=); do
...
+  setfacl -R -m m::rX ${SSL_CRTS_DIR}
+  setfacl -R -m u:${ETCD_UID}:rX ${SSL_CRTS_DIR}
+  setfacl -R -m g:${ETCD_UID}:rX ${SSL_CRTS_DIR}
```
Comment (superseb) [inline cert-deployer]: Guess this needs to be `ETCD_GID`

File: cert-deployer (hunk 1 above)
Comment (moelsayed) [reply cert-deployer]: I install the `acl` in the container, but not having ACLs enabled in the kernel is probably out of scope in our case. We don't catch any output in this script, so anything will be sent to the container's stdout/stderr.

**Comment by superseb** [comment]
LGTM

**Pull request merged**